converter/converters/pod/affinity.go

```diff
@@ -0,0 +1,398 @@
+package pod
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/golang/glog"
+	"github.com/koki/short/types"
+)
+
+func RevertAffinities(a []types.Affinity) (*v1.Affinity, error) {
+	node, pod, antiPod, err := splitAffinities(a)
+	if err != nil {
+		return nil, err
+	}
+
+	n, err := nodeAffinity(node)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := podAffinity(pod)
+	if err != nil {
+		return nil, err
+	}
+
+	ap, err := podAntiAffinity(antiPod)
+	if err != nil {
+		return nil, err
+	}
+
+	if n == nil && p == nil && ap == nil {
+		return nil, nil
+	}
+
+	return &v1.Affinity{
+		NodeAffinity:    n,
+		PodAffinity:     p,
+		PodAntiAffinity: ap,
+	}, nil
+}
+
+type PodAffinity struct {
+	Affinity   string
+	Topology   string
+	Namespaces []string
+}
+
+func splitAffinities(as []types.Affinity) (node []string, pod, antiPod []PodAffinity, err error) {
+	node = []string{}
+	pod = []PodAffinity{}
+	antiPod = []PodAffinity{}
+
+	for _, a := range as {
+		switch {
+		case len(a.NodeAffinity) > 0:
+			node = append(node, a.NodeAffinity)
+		case len(a.PodAffinity) > 0:
+			pod = append(pod, PodAffinity{
+				Affinity:   a.PodAffinity,
+				Topology:   a.Topology,
+				Namespaces: a.Namespaces,
+			})
+		case len(a.PodAntiAffinity) > 0:
+			antiPod = append(antiPod, PodAffinity{
+				Affinity:   a.PodAntiAffinity,
+				Topology:   a.Topology,
+				Namespaces: a.Namespaces,
+			})
+		default:
+			err = fmt.Errorf("unrecognized affinity %#v", a)
```

Comment:
Use util.TypeValueErrorf(type, format...)

Reply:
wait what? These are typed values... Why unrecognized affinity?

The parsing won't succeed if there are unrecognized values...

Reply:
That `default` case should be unreachable in normal use, but I'm marking it with an error just in case.